core/web/log_controller.go

```diff
@@ -0,0 +1,75 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"go.uber.org/zap/zapcore"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type LogPatchRequest struct {
+	Level      string `json:"level"`
+	SqlEnabled *bool  `json:"sqlEnabled"`
+}
+
+// Patch sets a log level and enables sql logging for the logger
+func (cc *LogController) Patch(c *gin.Context) {
+	request := &LogPatchRequest{}
+	if err := c.ShouldBindJSON(request); err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, err)
+		return
+	}
+
+	if request.Level == "" && request.SqlEnabled == nil {
+		jsonAPIError(c, http.StatusInternalServerError, fmt.Errorf("please set either logLevel or logSql as params in order to set the log level"))
+		return
+	}
+
+	if request.Level != "" {
+		var ll zapcore.Level
+		err := ll.UnmarshalText([]byte(request.Level))
+		if err != nil {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return
+		}
+		cc.App.GetStore().Config.Set("LOG_LEVEL", ll.String())
+		err = cc.App.GetStore().SetConfigStrValue("LogLevel", ll.String())
+		if err != nil {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return
+		}
+	}
+
+	if request.SqlEnabled != nil {
+		cc.App.GetStore().Config.Set("LOG_SQL", request.SqlEnabled)
+		err := cc.App.GetStore().SetConfigStrValue("LogSQLStatements", strconv.FormatBool(*request.SqlEnabled))
+		if err != nil {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return
+		}
+		cc.App.GetStore().SetLogging(*request.SqlEnabled)
+	}
+
+	// Set default logger with new configurations
+	logger.Default = cc.App.GetStore().Config.CreateProductionLogger()
```

Comment:
hm i think using the logger.SetLogger might be needed here instead. Though it might require some more improvements, considering the comment on it:

```
// DEPRECATED this method is deprecated because it leads to race conditions.
// Instead, you should fork the logger.Default instance to create a new logger
// for your module.
// Eg: logger.Default.Named("<my-package-name>")
```

Reply:
@PiotrTrzpil SetLogger just does the same thing `Default = <new logger>` so this will work, but I do think it probably is better to use SetLogger just to make it explicit. Re: race conditions, we only set this on boot so adding this single other write is OK. Refactoring the logger handling to use the forking (or some other cleaner solution) is a broader problem that I think should be handled all at once

Reply:
@connorwstein i see - yeah the flushing of logs that SetLogger does seems like should be done here, as this call can happen any time

Reply:
hm actually the flushing on the prev logger variable should likely be done after replacing the value on `logger.Default`

Reply:
spoke with @connorwstein  on this and we decided to stick with `SetLoggger` for now and will address the race condition in the package specific logging pr

Reply:
ok 👍🏼